src/backend/distributed/operations/repair_shards.c

```diff
@@ -287,7 +419,57 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 
 
 /*
- * master_move_shard_placement is a wrapper function for old UDF name.
+ * EnsureEnoughDiskSpaceForShardMove checks that there is enough space for
+ * shard moves of the given colocated shard list from source node to target node.
+ * It tries to clean up old shard placements to ensure there is enough space.
+ */
+static void
+EnsureEnoughDiskSpaceForShardMove(List *colocatedShardList,
+								  char *sourceNodeName, uint32 sourceNodePort,
+								  char *targetNodeName, uint32 targetNodePort)
+{
+	if (!CheckAvailableSpaceBeforeMove)
+	{
+		return;
+	}
+	uint64 colocationSizeInBytes = ShardListSizeInBytes(colocatedShardList,
+														sourceNodeName,
+														sourceNodePort);
+
+	uint32 connectionFlag = 0;
+	MultiConnection *connection = GetNodeConnection(connectionFlag, targetNodeName,
+													targetNodePort);
+	DeferredErrorMessage *temporaryError =
+		CheckSpaceConstraints(connection, colocationSizeInBytes);
+	if (temporaryError == NULL)
+	{
+		return;
+	}
+	RaiseDeferredError(temporaryError, WARNING);
+
+	/*
+	 * Only run shard cleanup when the first round of checks has failed
+	 * otherwise simply let the automatic cleanup do its job.
+	 *
+	 * NOTE: this tries to drop all marked shards, not only on the shard
+	 * that we're trying to move to.
+	 */
+	ereport(NOTICE, (errmsg("trying to drop old shards to resolve issue...")));
```

Comment:
`to resolve issue` sounds vague and can be confusing.  Also, can we return early if there are no `SHARD_STATE_TO_DELETE`?

And, if we have, we Can we tell something like: `found %d old shards on node XXX. Trying to drop old shards to allow the shard move operation to continue.`

Reply:
Makes sense,

I have one minor suggestion though, we have this at cluster level rather than node level ( of course we can get this at node level as well) but it is simpler to say something like the following:
```
ereport(NOTICE, (errmsg("found %d old shards, trying to drop them to allow the shard move operation to continue", numOfShardsToBeDeleted)));
```

Do you think that is enough apart from that we return early?